MetadataExtractor/Formats/Photoshop/DuckyReader.cs

```diff
@@ -67,7 +67,10 @@ public DuckyDirectory Extract(SequentialReader reader)
                         case DuckyDirectory.TagCopyright:
                         {
                             reader.Skip(4);
-                            directory.Set(tag, reader.GetString(length - 4, Encoding.BigEndianUnicode));
+                            length -= 4;
+                            if (length < 0)
+                                length = 0;
+                            directory.Set(tag, reader.GetString(length, Encoding.BigEndianUnicode));
```

Comment:
This change definitely fixes the bug. I would think that if we see an invalid length here we should add an error and stop processing rather than continuing (similar to lines 60-61).

Reply:
I changed the code according to your suggestion.